plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ContractDescriptionConversionVisitor.kt

```diff
@@ -196,13 +204,16 @@ class ContractDescriptionConversionVisitor(
         embeddedVarByIndex(parameterIndex)
 
     private fun KtValueParameterReference<ConeKotlinType, ConeDiagnostic>.getTargetParameter(data: ContractVisitorContext): FirBasedSymbol<*> {
-        return resolveByIndex(
+        return resolveByIndex<FirBasedSymbol<*>>(
             parameterIndex,
-            { data.functionContractOwner.receiverParameter!!.calleeSymbol }) { data.functionContractOwner.valueParameterSymbols[it] }
+            { data.functionContractOwner.receiverParameter!!.receiverSymbol }) { data.functionContractOwner.valueParameterSymbols[it] }
     }
 
     private fun embeddedVarByIndex(ix: Int): VariableEmbedding = resolveByIndex(ix, { signature.receiver!! }) { signature.params[it] }
 
+    private val FirReceiverParameter.receiverSymbol: FirBasedSymbol<*>
+        get() = session.symbolProvider.getSymbolByTypeRef<FirBasedSymbol<*>>(typeRef)!!
```

Comment:
Is this for extension or dispatch receivers?  Extension receivers are currently a mess in our code, I think we should leave them be until we have a good idea for how to handle them uniformly.

Reply:
I do agree, this code is for extension receivers.

Reply:
Probably best just to drop it for now then, I think Nils had already filed a bug about this ages ago.

Reply:
Yes, I remember the issue. But this getter is needed for making the `good_test/is_type_contract.kt` work. See this function here:

```kotlin
@OptIn(ExperimentalContracts::class)
fun Any.<!VIPER_TEXT!>isString<!>(): Boolean {
    contract {
        returns(true) implies (this@isString is String)
    }
    return this is String
}
```

Reply:
You can change the receiver to be a normal parameter.